command/ca/certificate.go

```diff
@@ -233,25 +229,38 @@ func (f *certificateFlow) getClient(ctx *cli.Context, subject, token string) (ca
 	root := ctx.String("root")
 	caURL := ctx.String("ca-url")
 
-	tok, err := jose.ParseSigned(token)
+	jwt, err := token.ParseInsecure(tok)
 	if err != nil {
 		return nil, errors.Wrap(err, "error parsing flag '--token'")
 	}
-	var claims tokenClaims
-	if err := tok.UnsafeClaimsWithoutVerification(&claims); err != nil {
-		return nil, errors.Wrap(err, "error parsing flag '--token'")
-	}
-	if strings.ToLower(claims.Subject) != strings.ToLower(subject) {
-		return nil, errors.Errorf("token subject '%s' and CSR CommonName '%s' do not match", claims.Subject, subject)
+	switch jwt.Payload.Type() {
```

Comment:
I don't understand the necessity to perform these checks.

The only place I see calling `getClient` uses the subject from the `CSR` (line 316) which was generated from the token (in the context of the certificate action). In the CreateSignRequest method you populate the CN field of the CSR (from the token) and then in this method you are verifying that the value from the CSR matches the value from the token. Unless I'm missing something (which I probably am) I can't see the reason to check a value that you just set. 

I'm a bit confused.

Reply:
We're not generating the certificate from the token if you use `step ca sign`

Reply:
I see. So the CertificateFlow is reused for other actions. In this case the checks are redundant, but they won't always be.